### PR TITLE
Add owners/approvers aliases for sig-auth subprojects

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

* Adds owner aliases for approvers and reviewers for the sig-auth subprojects.
* References those aliases from owners files in the packages related to that subproject.
* Where there were existing approvers/reviewers, hoisted them into the alias lists.
* Where there were not existing approvers/reviewers (for the audit and encryption-at-rest subprojects), populated lists with kubernetes members in the committers lists relevant to the subproject (added as TMP commits in https://github.com/liggitt/kubernetes/commits/sig-auth-subprojects-commits for inspection)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

xref https://github.com/kubernetes/community/pull/2525

**Special notes for your reviewer**:

OWNERS and COMMITS files generated for each package with https://gist.github.com/liggitt/642a1c5ae1eb16262729a248fe6869ff

```release-note
NONE
```

/hold
for review

will mention on the sig-auth mailing list and in the [subprojects pr](https://github.com/kubernetes/community/pull/2525)

### CODE REVIEW

**Comment by liggitt** [comment]
cc @kubernetes/sig-auth-misc for review
cc @kubernetes/steering-committee for visibility

File: pkg/apis/authentication/OWNERS [hunk 1]
```diff
@@ -1,9 +1,6 @@
 reviewers:
-- liggitt
-- lavalamp
-- wojtek-t
-- deads2k
-- sttts
-- mbohlool
-- jianhuiz
-- enj
+- sig-auth-authentication-approvers
```
Comment (mikedanese) [inline pkg/apis/authentication/OWNERS]: s/authentication/authenticators ?

**Comment by dims** [comment]
"added as TMP commits for inspection, will drop before merging" << Big +1

File: pkg/apis/authentication/OWNERS (hunk 1 above)
Comment (liggitt) [reply pkg/apis/authentication/OWNERS]: fixed

**Comment by deads2k** [comment]
I don't see any surprises.

**Comment by liggitt** [comment]
moved TMP commits containing commit history to https://github.com/liggitt/kubernetes/commits/sig-auth-subprojects-commits to clean this branch up for merge

added an owners file for `pkg/kubectl/cmd/auth/OWNERS` for the authorizers subproject based on review in https://github.com/kubernetes/community/pull/2525/files#r229072570

as this is intentionally as close to a no-op as possible, would like to get this merged and iterate on the subproject membership

File: OWNERS_ALIASES [hunk 2]
```diff
@@ -1,4 +1,123 @@
...
+    - mml
+    - ncdc
+    - nikhiljindal
+    - smarterclayton
+    - sttts
+    - thockin
+    - wojtek-t
+
+  sig-auth-certificates-approvers:
+    - mikedanese
```
Comment (mikedanese) [inline OWNERS_ALIASES]: This one is non-alphabetic.

File: OWNERS_ALIASES [hunk 3]
```diff
@@ -1,4 +1,123 @@
...
+    - liggitt
+    - sakshamsharma
+    - smarterclayton
+    - wojtek-t
+
+  sig-auth-node-isolation-approvers:
+    - deads2k
+    - liggitt
+    - tallclair
+    - mikedanese
```
Comment (mikedanese) [inline OWNERS_ALIASES]: This one is non-alphabetic.

**Review by mikedanese — COMMENTED** [review]
lgtm
/approve

**Comment by liggitt** [comment]
> This one is non-alphabetic.

fixed

File: plugin/pkg/auth/authorizer/node/OWNERS [hunk 4]
```diff
@@ -1,8 +1,7 @@
 approvers:
-- tallclair
-- liggitt
-- deads2k
+- sig-auth-node-restriction-approvers
```
Comment (tallclair) [inline plugin/pkg/auth/authorizer/node/OWNERS]:
```suggestion
- sig-auth-node-isolation-approvers
```
Same below.

xref: https://github.com/kubernetes/test-infra/issues/8838#issuecomment-436075246

**Review by tallclair — COMMENTED** [review]
Thanks for working on this. Please fix the node-restriction alias, otherwise /lgtm

File: OWNERS_ALIASES [hunk 5]
```diff
@@ -1,4 +1,123 @@
...
+    - deads2k
+    - dims
+    - enj
+    - erictune
+    - jianhuiz
+    - krousey
+    - lavalamp
+    - liggitt
+    - mbohlool
+    - mml
```
[the remark this answers is not in the document]
Comment (liggitt) [reply OWNERS_ALIASES]: I left these as-is for now, can discuss what we want this to look like (might differ per person)

**Comment by enj** [comment]
/lgtm
/approve

I think this is a good place to start.  We can iterate on this as needed.

**Comment by liggitt** [comment]
> We suggest the following additional approver: thockin

/assign @thockin
for OWNERS_ALIASES additions
has lgtm from the sig-auth chairs

**Pull request merged**